src/api/spec/models/package_spec.rb

```diff
@@ -98,4 +104,74 @@
       end
     end
   end
+
+  context '#maintainers' do
+    before do
+      User.current = user
+    end
+
+    it 'returns an array with user objects to all maintainers for a package' do
+      # first of all, we add a user who is not a maintainer but a bugowner
+      # he/she should not be recognized by package.maintainers
+      package.add_user(other_user2, Role.find_by_title!("bugowner"))
+      package.save!
+
+      # we expect to get one user object in an array
+      package.add_user(user, maintainer_role)
+      package.save!
+
+      expect(package.maintainers).to match_array([user])
+
+      # we expect the previous user and the following user to be in that array
+      package.add_user(other_user, maintainer_role)
+      package.save!
+
+      expect(package.maintainers).to match_array([other_user, user])
+    end
+
+    it 'resolves groups properly' do
+      # groups should be resolved and only their assigned users should be in the
+      # returning array
+      group.add_user(other_user)
+      group.add_user(other_user2)
+
+      # add a group to the package what is not a maintainer to make sure it'll
+      # be ignored when calling package.maintainers
+      group_bugowner.add_user(other_user3)
+
+      package.add_group(group_bugowner, Role.find_by_title!("bugowner"))
```

Comment:
Instead of doing:
```ruby
package.add_group(group_bugowner, Role.find_by_title!("bugowner"))
package.save
```

Just create and use  a factory `relationship_package_group_as_bugowner `:
```ruby
create(:relationship_package_group_as_bugowner, group: group_bugowner, package: package)
```
Here you have the factory:
```ruby
# In the file src/api/spec/factories/relationship.rb
...
factory :relationship_package_group do
  package
  group

  factory :relationship_package_group_as_bugowner do
    role { Role.find_by_title('bugowner') }
  end
end
...
```